chainstate/src/detail/block_index_history_iter.rs

```diff
@@ -64,17 +64,18 @@ mod tests {
     use super::*;
 
     #[test]
-    fn mainnet_creation() {
+    fn history_iteration() {
         common::concurrency::model(|| {
             let chain_config = Arc::new(create_unit_test_config());
             let storage = Store::new_empty().unwrap();
-            let mut chainstate = Chainstate::new(chain_config.clone(), storage, None).unwrap();
+            let mut chainstate =
+                Chainstate::new(chain_config.clone(), storage, None, None).unwrap();
 
             // put three blocks in a chain after genesis
             let block1 = Block::new(
                 vec![],
                 Some(Id::new(&chainstate.chain_config.genesis_block_id().get())),
-                time::get() as u32,
+                time::get().as_secs() as u32,
```

Comment:
It isn't directly related to this PR,  but maybe we want for `Block::new` to accept `Duration` instead of `u32` in order to get rid of all these `time::get().as_secs() as u32`?

Reply:
That's going to be the next step. This is just the beginning. There will be a BlockTimestamp type that will solve this problem.

One step at a time ;-)